Automattic/wp-calypso
# Comments: move active reply comment to Redux

The **active reply comment** is the comment with a reply textbox shown beneath it on the front end:

![screen shot 2017-09-29 at 17 22 15](https://user-images.githubusercontent.com/17325/31025669-5019baee-a53b-11e7-9740-8ea6f24078bb.png)

We currently keep track of the active reply comment in `PostCommentList`'s component state. This works fine, but it makes it tricky to change the active reply comment in response to another Redux action. I ran into this while working on https://github.com/Automattic/wp-calypso/pull/18356.

This PR moves the active reply comment state to Redux.

### To test

In a comment thread:
* ensure that clicking 'reply' makes the comment textbox appear under the current comment
* ensure that clicking 'cancel reply' returns the comment textbox to the top level
* ensure that submitting a comment successfully returns the comment textbox to the top level
* ensure that the active comment is not preserved when revisiting a page (it should be reset on mount)

Here's a good comments thread to test with:

http://calypso.localhost:3000/read/feeds/40474296/posts/1385874605
